yjit/src/backend/ir.rs

```diff
@@ -2002,6 +2004,16 @@ impl Assembler {
     }
 }
 
+/// Macro to use format! for asm.comment, which skips a format! call
+/// when disasm is not supported.
+macro_rules! asm_comment {
+    ($asm:expr, $($fmt:tt)*) => {
+        #[cfg(feature = "disasm")]
+        $asm.comment(&format!($($fmt)*));
+    };
+}
+pub(crate) use asm_comment;
```

Comment:
AFAIK the asm_comment macro shouldn't be necessary. With the cfg! check you put above, asm.comment becomes an empty method if the disasm feature is not enabled.

Reply:
It's necessary because the `format!` macro allocates a `String` at the call site before entering the noop function. We have more certainty that we're skipping the allocation with this macro.